### PR TITLE
va-back-to-top: Change button to anchor link

### PR DESCRIPTION
**Pull request by jamigibbs**
## Chromatic

https://899-back-to-top-a11y--60f9b557105290003b387cd5.chromatic.com/?path=/docs/components-va-back-to-top--default

## Description

Sketch: https://www.sketch.com/s/73212626-acc4-47e6-acd8-d4e2f5a98372

Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/899
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/58413
Related https://github.com/department-of-veterans-affairs/va.gov-team/issues/60812

## Testing done
Storybook
Voiceover
vets website locally

## Screenshots

Screenshot 2023-07-13 at 1 10 14 PM

Screenshot 2023-07-13 at 1 10 10 PM

Screenshot 2023-07-13 at 1 10 05 PM

Screenshot 2023-07-13 at 1 55 42 PM

Vets Website Demo

https://github.com/department-of-veterans-affairs/component-library/assets/872479/f82c896a-6f12-416e-97df-6a8520399024

## Acceptance criteria
- [ ] va-back-to-top has been updated from a button to a link

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-back-to-top/va-back-to-top.tsx [hunk 1]
```diff
@@ -86,10 +87,13 @@ export class VaBackToTop {
           <div
             class={classnames({ docked: this.isDocked, reveal: this.revealed })}
           >
-            <button type="button" onClick={this.navigateToTop.bind(this)}>
-              <i aria-hidden="true" role="img"></i>
-              <span>Back to top</span>
-            </button>
+            <a href="#ds-back-to-top" onClick={this.navigateToTop.bind(this)}>
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-back-to-top/va-back-to-top.tsx:90]: In order pass the accessibility Axe check, the link would need to be in one of the following patterns which is why I added the `href="#ds-back-to-top"`: 

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/anchor-is-valid.md#succeed

I think what I would have liked to do here is to use the href as an actual link to an element with that ID. I experimented with doing something like programmatically adding a div directly after the `body` tag like this:

```js
const backToTopEl = document.createElement('div');
backToTopEl.id = 'ds-back-to-top';
document.body.parentNode.insertBefore(backToTopEl, document.body);
```

But the click is doing more than just navigating. It's adding a focus to the first `h1` on the page and this also didn't work very well in an iframe (like Storybook). 

So I'm adding `href="#ds-back-to-top"` to make the link valid but it doesn't actually function in the way it appears. The alternative was to add `// eslint-disable-next-line anchor-is-valid` above the line.

**Review by powellkerry — APPROVED** [review]

**Comment by powellkerry** [comment]
LGTM

**Comment by babsdenney** [comment]
This looks great and matches the designs Matt created. Design approved!

**Comment by humancompanion-usds** [comment]
The color of the arrow at mobile doesn't appear to be correct. Is that the visited color perchance? This element shouldn't have a different visited link color (should stay the same blue). 

back-to-top-at-mobile

**Comment by jamigibbs** [comment]
> The color of the arrow at mobile doesn't appear to be correct. Is that the visited color perchance? This element shouldn't have a different visited link color (should stay the same blue).
> 
> back-to-top-at-mobile

Good eye. Yes, that looks like the browser's default visited color. I'll get that updated. 👍

**Pull request merged**